CodingChallenge/MusicSearch/MusicSearchRouter.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  MusicSearchRouter.swift
+//  CodingChallenge
+//
+//  Created by Edgar Barocio on 6/22/19.
+//  Copyright (c) 2019 Edgar Barocio. All rights reserved.
+//
+//  This file was generated by the Clean Swift Xcode Templates so
+//  you can apply clean architecture to your iOS and Mac projects,
+//  see http://clean-swift.com
+//
+
+import UIKit
+
+@objc protocol MusicSearchRoutingLogic {
+    //func routeToSomewhere(segue: UIStoryboardSegue?)
+}
+
+protocol MusicSearchDataPassing {
+    var dataStore: MusicSearchDataStore? { get }
+}
+
+class MusicSearchRouter: NSObject, MusicSearchRoutingLogic, MusicSearchDataPassing {
+    weak var viewController: MusicSearchViewController?
+    var dataStore: MusicSearchDataStore?
+    
+    // MARK: Routing
+    
+    //func routeToSomewhere(segue: UIStoryboardSegue?)
+    //{
+    //  if let segue = segue {
+    //    let destinationVC = segue.destination as! SomewhereViewController
+    //    var destinationDS = destinationVC.router!.dataStore!
+    //    passDataToSomewhere(source: dataStore!, destination: &destinationDS)
+    //  } else {
+    //    let storyboard = UIStoryboard(name: "Main", bundle: nil)
+    //    let destinationVC = storyboard.instantiateViewController(withIdentifier: "SomewhereViewController") as! SomewhereViewController
+    //    var destinationDS = destinationVC.router!.dataStore!
+    //    passDataToSomewhere(source: dataStore!, destination: &destinationDS)
+    //    navigateToSomewhere(source: viewController!, destination: destinationVC)
+    //  }
+    //}
+    
+    // MARK: Navigation
+    
+    //func navigateToSomewhere(source: MusicSearchViewController, destination: SomewhereViewController)
+    //{
+    //  source.show(destination, sender: nil)
+    //}
+    
+    // MARK: Passing data
+    
+    //func passDataToSomewhere(source: MusicSearchDataStore, destination: inout SomewhereDataStore)
+    //{
+    //  destination.name = source.name
+    //}
+}
```

Comment:
Delete commented code unless you want to show knowledge of segues (but not use segues).